compiler-rt/lib/asan/asan_allocator.cpp

```diff
@@ -732,7 +733,8 @@ struct Allocator {
     if (!AtomicallySetQuarantineFlagIfAllocated(m, ptr, stack)) return;
 
     if (m->alloc_type != alloc_type) {
-      if (atomic_load(&alloc_dealloc_mismatch, memory_order_acquire)) {
+      if (atomic_load(&alloc_dealloc_mismatch, memory_order_acquire) &&
+          !IsStackTraceSuppressed(stack)) {
```

Comment:
This looks very random.
Why ReportAllocTypeMismatch and not ReportNewDeleteTypeMismatch?

I propose to move IsStackTraceSuppressed into Report* functions.
WDYT?

Do do so, some refactoring is needed, some Report* functions don't accept stack,
but they generate it internally anyway. We need to change that, and than move existing IsStackTraceSuppressed into Report* function.

I don't have preference if we do that before, or after the current PR.